packages/ui/src/components/ListStats.vue

```diff
@@ -0,0 +1,99 @@
+<template>
+  <div>
+    <div v-if="primaryDataType === 'bufficorns'">
+      <BufficornGlobalData
+        v-for="(bufficorn, index) in sortedBufficornsData"
+        :index="index"
+        :key="bufficorn.name"
+        :name="bufficorn.name"
+        :vigor="bufficorn.vigor"
+        :agility="bufficorn.agility"
+        :speed="bufficorn.speed"
+        :coolness="bufficorn.coolness"
+      />
+    </div>
+    <div v-if="primaryDataType === 'players'">
+      <PlayerGlobalData
+        v-for="(player, index) in sortedPlayersData"
+        :key="player.username"
+        :index="index"
+        :name="player.username"
+        :position="player.position"
+        :score="player.points"
+      />
+    </div>
+    <div v-if="primaryDataType === 'ranches'">
+      <RancheGlobalData
+        v-for="(ranch, index) in sortedRanchesData"
+        :index="index"
+        :score="ranch.score"
+        :name="ranch.name"
+        :key="ranch.name"
+      />
+    </div>
+  </div>
+</template>
+
+<script>
+import { useStore } from '@/stores/player'
+import { computed, onBeforeMount, nextTick } from 'vue'
+export default {
+  props: {
+    primaryDataType: {
+      type: String,
+      required: true
+    },
+    secondaryDataType: {
+      type: String,
+      required: true
+    }
```

Comment:
Can we use more idiomatic meaningful?

```suggestion
    gameEntity: {
      type: String,
      required: true
    },
    
    // filterBy, attribute or trait
    filterBy: {
      type: String,
      required: true
    }
```